code/modules/implant_hacking/board.dm

```diff
@@ -0,0 +1,288 @@
+
+		if(xcord < board_size && ((EAST in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += EAST
+
+		if(ycord > 1 && ((NORTH in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += NORTH
+
+		if(ycord < board_size && ((SOUTH in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += SOUTH
+
+		board[xcord][ycord].connected = FALSE
+
+		for(var/dir in possible_directions)
+
+			switch(dir)
+
+				if(NORTH)
+
+					dirlist = board[xcord][ycord-1].get_dir_val_list()
+
+					if(board[xcord][ycord-1].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(SOUTH in dirlist)
+						if(!board[xcord][ycord-1].visited)
+							queued_coord_list += list(list(xcord,ycord-1))
+						board[xcord][ycord].connected = TRUE
+
+				if(SOUTH)
+
+					dirlist = board[xcord][ycord+1].get_dir_val_list()
+
+					if(board[xcord][ycord+1].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(NORTH in dirlist)
+						if(!board[xcord][ycord+1].visited)
+							queued_coord_list += list(list(xcord,ycord+1))
+						board[xcord][ycord].connected = TRUE
+
+				if(WEST)
+
+					dirlist = board[xcord-1][ycord].get_dir_val_list()
+
+					if(board[xcord-1][ycord].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(EAST in dirlist)
+						if(!board[xcord-1][ycord].visited)
+							queued_coord_list += list(list(xcord-1,ycord))
+						board[xcord][ycord].connected = TRUE
+
+				if(EAST)
+
+					dirlist = board[xcord+1][ycord].get_dir_val_list()
+
+					if(board[xcord+1][ycord].finish)
+						return TRUE
+
+					if(WEST in dirlist)
+						if(!board[xcord+1][ycord].visited)
+							queued_coord_list += list(list(xcord+1,ycord))
+						board[xcord][ycord].connected = TRUE
+
+		board[xcord][ycord].visited = TRUE
+
+	return FALSE
+/**
+ * Simple holder of information and some procs relating to this information
+ *
+ * Holds relevant information about a single cell of the board.
+ */
+/datum/hacking_minigame_piece
+	var/rotation = 0
+	var/pass_in = NONE
+	var/pass_out = NONE
+	var/visited = FALSE
+	var/start = FALSE
+	var/finish = FALSE
+	var/connected = FALSE
+
+	var/datum/hacking_minigame/game
+
+/datum/hacking_minigame_piece/New(datum/hacking_minigame/_game)
+	. = ..()
+	game = _game
+
+///Returns value of pass_in + pass_out adjusted for rotation
+/datum/hacking_minigame_piece/proc/get_dir_val()
+	if(pass_in != NONE && pass_out != NONE)
+		return angle2dir(dir2angle(pass_in)+rotation*90) + angle2dir(dir2angle(pass_out)+rotation*90)
+	return 0
+
+///Returns a list of possible directions this cell can connect to
+/datum/hacking_minigame_piece/proc/get_dir_val_list()
+	return finish == TRUE || start == TRUE ? GLOB.cardinals.Copy() : list(pass_in != NONE ? angle2dir(dir2angle(pass_in)+rotation*90) : 0 , pass_out != NONE ? angle2dir(dir2angle(pass_out)+rotation*90) : 0 )
```

Comment:
this proc is hell to read as a one-liner, please split this up into multiple lines so its clearer

Reply:
look 3 down